communication/CMakeLists.txt

```diff
@@ -13,22 +13,32 @@ if(${BUILD_COMM_TCP_TLS})
 endif()
 
 add_library(bftcommunication ${bftcommunication_src})
+add_library(bftcommunication_shared SHARED ${bftcommunication_src})
 target_include_directories(bftcommunication PUBLIC include)
 target_link_libraries(bftcommunication PUBLIC util)
+target_include_directories(bftcommunication_shared PUBLIC include)
+target_link_libraries(bftcommunication_shared PUBLIC util_shared)
 
 if(${BUILD_COMM_TCP_PLAIN} OR ${BUILD_COMM_TCP_TLS})
-    set(Boost_USE_STATIC_LIBS ON) # only find static libs
+    set(Boost_USE_STATIC_LIBS OFF) # only find static libs
```

Comment:
Change the comment to 'shared lib'

Reply:
Changed the comment to both things, because it's also not just shared